synapse/media/preview_html.py

```diff
@@ -381,8 +405,8 @@ def parse_html_description(tree: "etree.Element") -> Optional[str]:
 
 
 def _iterate_over_text(
-    tree: Optional["etree.Element"],
-    tags_to_ignore: Set[Union[str, "etree.Comment"]],
+    tree: Optional["etree._Element"],
+    tags_to_ignore: Set[object],
```

Comment:
`etree.Comment` isn't a type (since it is a function), and (as the comment above states) using `etree._Comment` is not the correct thing to do.